sample_keyprovider/README.md

```diff
@@ -32,24 +73,42 @@ vim ocicrypt.conf:
 Copy the image you want to encrypt to your current directory. This example uses a *busybox* image:
 
 ```
-skopeo copy docker://busybox oci:busybox
+OCICRYPT_KEYPROVIDER_CONFIG=ocicrypt.conf skopeo copy --encryption-key provider:attestation-agent:<parameters> docker://busybox oci:busybox_encrypted
 ```
+After encryption, it can be seen that busybox-encrypted is generated in the current directory.
 
-Encrypt the container image. You can pass in the special parameters required by the encryption module through the command line parameters of skopeo (here, the string "test" is taken as an exemplary special parameter): 
+Or we can directly push the encrypted image to the remote image registry
 
 ```
-OCICRYPT_KEYPROVIDER_CONFIG=ocicrypt.conf skopeo copy --insecure-policy --encryption-key provider:attestation-agent:test oci:busybox oci:busybox-encrypted
+OCICRYPT_KEYPROVIDER_CONFIG=ocicrypt.conf skopeo copy --encryption-key provider:attestation-agent:<parameters> docker://busybox docker://docker.io/myrepo/busybox:encrypted
 ```
 
-After encryption, it can be seen that busybox-encrypted is generated in the current directory.
+### Parameters
+
+The `<parameters>` in skopeo command is a key-value list separated by double colons. Here are the defined keys:
+- `sample`: Not required. Either `true` or `false`. If not set, use `false`. This value indicates whether the hardcoded encryption key is used. This works the same way as `sample keyprovider`.
+- `keyid`: Required if `sample` is not enabled. It is a KBS Resource URI, s.t. `kbs://<kbs-addr>/<repo>/<type>/<tag>`. When decryption occurs, the `keyid` value is used to index the KEK.
+- `keypath`: Required if `sample` is not enabled. A local filesystem path, absolute path recommended. Specify the KEK to encrypted the image in local filesystem. KEK will be read from fs and then used to encrypt the image. This key's length must be 32 bytes.
+- `algorithm`: Not required. Indicate the encryption algorithm used. Either `A256GCM` or `A256CTR`. If not provided, use `A256GCM` by default as it is AEAD scheme.
 
-### Decryption
+### Examples
+
+- Use key `key1`, and keyid `kbs://example.org/test-repo/key/key_id1` to encrypt an image. In this way sample is disabled, and will use A256GCM (AES-256-GCM).
```

Comment:
These parameters seem inconsistent. It says use "key `key1`", but there is no parameter called key. Is that supposed to be `keypath`? Shouldn't the keypath be something like `aa-offline_fs_kbc-keys.json`?